ci/hermetic/generate_artifacts_lock_file.py

```diff
@@ -0,0 +1,151 @@
+lock file.
+
+For each row in the input CSV, it expects the following columns:
+- 'download_url': The URL from which to download the artifact.
+- 'checksum': The name of the hash algorithm (e.g., 'sha256', 'sha512')
+- 'filename': The name of the output file.
+
+The script performs the following actions:
+1. Reads the input CSV file row by row.
+2. For each artifact, it downloads the content from the 'download_url'.
+3. It calculates the checksum of the downloaded content using the specified
+   hash algorithm.
+4. It updates the 'checksum' field in the row to include both the algorithm
+   and the calculated checksum in the format '<algorithm>:<checksum>'.
+5. Writes the transformed data, along with a metadata header, into a YAML file.
+
+The generated YAML file serves as a lock file, ensuring that artifacts can
+be validated against their recorded checksums.
+
+Usage:
+    ./generate_artifacts_lock_file.py <input_csv_fpath> <output_yaml_fpath>
+
+Example:
+    ./generate_artifacts_lock_file.py artifacts.in.csv artifacts.lock.yaml
+"""
+
+
+def csv_to_yaml(csv_filepath: str, yaml_filepath: str) -> None:
+    """
+    Converts a CSV file to a YAML file.
+
+    Each row in the CSV becomes an item in a YAML list.
+    The CSV column names become keys in the dictionary for each item.
+
+    Args:
+        csv_filepath (str): The path to the input CSV file.
+        yaml_filepath (str): The path to the output YAML file.
+    """
+    data = {'metadata': {'version': '1.0'}, 'artifacts': []}
+    try:
+        with open(csv_filepath, mode='r', newline='',
+                  encoding='utf-8') as csv_file:
+            csv_reader = csv.DictReader(csv_file)
+            if csv_reader.fieldnames is None:
+                print(f"Error: CSV file '{csv_filepath}' "
+                      "is empty or has no header.", file=sys.stderr)
+                return
+
+            for row in csv_reader:
+                # We calculate the checksum of the file using the hash
+                # algorithm, and store the value in the desired format
+                # '<hash_algorithm>:<checksum>'.
+                row['checksum'] = "{}:{}".format(
+                    row['checksum'],
+                    calculate_checksum(row['download_url'], row['checksum'])
+                )
+                data['artifacts'].append(row)
+
+        with open(yaml_filepath, mode='w', encoding='utf-8') as yaml_file:
+            yaml_file.write(
+                "# This file was generated automatically by '{}' "
+                "from the content\n"
+                "# of artifacts.in.csv. Do not edit this file, "
+                "edit artifacts.in.csv instead."
+                "\n#\n#           ANY CHANGES MADE HERE WILL BE LOST!\n#\n"
+                "# In order to validate locally the data, please refer to "
+                "the README file located at the root directory of this "
+                "script.\n".format(sys.argv[0])
+            )
```

Comment:
Use a multi-line string:
```suggestion
yaml_file.write(
    """# This file was generated automatically by '{}' from the content
    # of artifacts.in.csv. Do not edit this file, edit artifacts.in.csv instead.
    #
    #           ANY CHANGES MADE HERE WILL BE LOST!
    #
    # In order to validate locally the data, please refer to the README file 
    # located at the root directory of this script.""".format(sys.argv[0])
)
```

Reply:
This would violate https://www.flake8rules.com/rules/E501.html when indenting properly.
I used your suggestion initially but had to use current definition to not violate E501.
One solution would be define your multiline suggestion as a global variable (i.e no indent).

Reply:
I removed the use of this warning message as the python script now updates directly the `artifacts.lock.yaml` file